### PR TITLE
[#388] respect defaults from cr configured probe, avoid unnecessary r…

### PR DESCRIPTION
**Pull request by gtully**
…econcile

### CODE REVIEW

File: controllers/activemqartemis_reconciler.go [hunk 1]
```diff
@@ -93,16 +93,6 @@ type ValueInfo struct {
 	Internal bool //if true put this value to the internal secret
 }
 
-type ActiveMQArtemisIReconciler interface {
```
[the remark this answers is not in the document]
Comment (gtully) [reply controllers/activemqartemis_reconciler.go:96]: it is just a tidy up. does not really warrant a PR, but I agree it is cleaner to keep it out of this PR. will sort.

File: controllers/activemqartemis_reconciler.go [hunk 2]
```diff
@@ -2006,6 +1985,28 @@ func configureReadinessProbe(container *corev1.Container, probe *corev1.Probe) *
 	return readinessProbe
 }
 
+func applyNonDefaultedValues(readinessProbe *corev1.Probe, probeFromCr *corev1.Probe) {
+	//copy the probe, but only for non default (init values)
+	if probeFromCr.InitialDelaySeconds > 0 {
```
[the remark this answers is not in the document]
Comment (gtully) [reply controllers/activemqartemis_reconciler.go:1990]: it is optional, so the implicit default is 0. It is just not called out in the doc.

File: controllers/suite_test.go [hunk 3]
```diff
@@ -615,6 +615,14 @@ func MatchCapturedLog(pattern string) (matched bool, err error) {
 	return regexp.Match(pattern, logBuffer.Bytes())
 }
 
+func FinaAllFromCapturedLog(pattern string) []string {
```
[the remark this answers is not in the document]
Comment (gtully) [reply controllers/suite_test.go:618]: good find!

File: controllers/activemqartemis_controller_test.go [hunk 4]
```diff
@@ -1591,6 +1591,51 @@ var _ = Describe("artemis controller", func() {
...
+
+		It("verify mimimal updates via unequal", func() {
+
+			if os.Getenv("USE_EXISTING_CLUSTER") == "true" {
+
+				crd := generateArtemisSpec(defaultNamespace)
+				crd.Spec.DeploymentPlan.Size = 1
+
+				crd.Spec.DeploymentPlan.ReadinessProbe = &corev1.Probe{
+					InitialDelaySeconds: 1,
```
[the remark this answers is not in the document]
Comment (gtully) [reply controllers/activemqartemis_controller_test.go:1603]: same as above, b/c it is optional - 0 is the implicit default

**Comment by gtully** [comment]
Thanks Dom

**Review by brusdev — APPROVED** [review]

**Pull request merged**